faster_whisper/transcribe.py

```diff
@@ -775,20 +848,30 @@ def find_alignment(
             for i, j in zip(word_boundaries[:-1], word_boundaries[1:])
         ]
 
-        # hack: ensure the first and second word is not longer than twice the median word duration.
+        # hack: truncate long words at the start of a window and the start of a sentence.
         # a better segmentation algorithm based on VAD should be able to replace this.
         word_durations = end_times - start_times
         word_durations = word_durations[word_durations.nonzero()]
         if len(word_durations) > 0:
             median_duration = np.median(word_durations)
             max_duration = median_duration * 2
-            if len(word_durations) >= 2 and word_durations[1] > max_duration:
-                boundary = max(end_times[2] / 2, end_times[2] - max_duration)
-                end_times[0] = start_times[1] = boundary
-            if (
-                len(word_durations) >= 1
-                and end_times[0] - start_times[0] > max_duration
-            ):
+            sentence_end_marks = ".。!！?？"
+            # ensure words at sentence boundaries are not longer than twice the median
```

Comment:
It would be helpful to have an explanation why we need to ensure that.

Reply:
Added as suggested.